gc.c

```diff
@@ -9745,7 +9743,6 @@ gc_move(rb_objspace_t *objspace, VALUE scan, VALUE free, size_t src_slot_size, s
     return (VALUE)src;
 }
 
-#if GC_CAN_COMPILE_COMPACTION
```

Comment:
I think this condition should be kept.

Reply:
We can't keep this because `gc_sort_heap_by_empty_slots` is used in `gc_verify_compaction_references`, so keeping this will fail to compile.

Reply:
https://github.com/nobu/ruby/tree/unused-code-when-CANNOT_COMPILE_COMPACTION

Reply:
That patch looks good.